src/ui/components/address-action/TransactionConfirmationView/TransactionConfirmationView.tsx

```diff
@@ -96,14 +99,25 @@ export function TransactionConfirmationView({
         method="dialog"
         style={{ display: 'flex', flexDirection: 'column' }}
       >
-        <TransactionSimulation
-          showApplicationLine={showApplicationLine}
-          localAllowanceQuantityBase={localAllowanceQuantityBase}
-          onOpenAllowanceForm={onOpenAllowanceForm}
-          address={wallet.address}
-          transaction={transaction}
-          txInterpretQuery={txInterpretQuery}
-        />
+        {transaction.evm ? (
+          <TransactionSimulation
+            showApplicationLine={showApplicationLine}
+            localAllowanceQuantityBase={localAllowanceQuantityBase}
+            onOpenAllowanceForm={onOpenAllowanceForm}
+            address={wallet.address}
+            transaction={transaction.evm}
+            txInterpretQuery={txInterpretQuery}
+          />
+        ) : (
+          <AddressActionComponent
+            address={wallet.address}
+            showApplicationLine={true}
+            addressAction={parseSolanaTransaction(
```

Comment:
Do I understand it right that this logic will be moved to transaction simulations very soon?